src/Android/Avalonia.Android/Platform/SkiaPlatform/InvalidationAwareSurfaceView.cs

```diff
@@ -14,12 +14,13 @@ internal abstract class InvalidationAwareSurfaceView : SurfaceView, ISurfaceHold
     {
         bool _invalidateQueued;
         private bool _isDisposed;
+        private bool _isSurfaceValid;
         readonly object _lock = new object();
         private readonly Handler _handler;
 
         internal event EventHandler? SurfaceWindowCreated;
 
-        IntPtr IPlatformHandle.Handle => Holder?.Surface?.Handle is { } handle ?
+        IntPtr IPlatformHandle.Handle => _isSurfaceValid && Holder?.Surface?.Handle is { } handle ?
             AndroidFramebuffer.ANativeWindow_fromSurface(JNIEnv.Handle, handle) :
```

Comment:
This leaks memory, BTW. `ANativeWindow_fromSurface` increases the reference counter for returned ANativeWindow object.

It's out of the scope of the current PR but should be addressed at some point